cc/cc.go

```diff
@@ -0,0 +1,149 @@
+package cc
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil" // nolint:staticcheck
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/go-co-op/gocron"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/node-real/go-pkg/log"
+	"github.com/shopspring/decimal"
+
+	"github.com/bnb-chain/bsc-staking-indexer/store"
+)
+
+const url = "https://ccalernow.bk.nodereal.cc/sendalerts"
+
+type CC struct {
+	cfg       Config
+	store     store.Store
+	scheduler *gocron.Scheduler
+}
+
+func New(cfg Config, store store.Store) *CC {
+	cc := &CC{
+		cfg:       cfg,
+		store:     store,
+		scheduler: gocron.NewScheduler(time.UTC),
+	}
+
+	if _, err := cc.scheduler.Month(1).Do(func() {
+		cc.ComputeAndSend()
+	}); err != nil {
+		log.Debugw("error while setting up scheduler", "err", err)
+	}
+
+	return cc
+}
+
+func (c *CC) Start() {
+	c.scheduler.StartAsync()
+}
+
+func (c *CC) ComputeAndSend() {
+	var (
+		ctx   = context.Background()
+		year  = time.Now().Year()
+		month = time.Now().Month()
+		from  time.Time // the beginning of last month
+		to    time.Time // the beginning of this month
+	)
+
+	if month == 1 {
+		from = time.Date(year-1, 12, 0, 0, 0, 0, 0, time.UTC)
+	} else {
+		from = time.Date(year, month-1, 0, 0, 0, 0, 0, time.UTC)
+	}
+
+	to = time.Date(year, month, 0, 0, 0, 0, 0, time.UTC)
+
+	var (
+		commissionReward     = decimal.Zero
+		selfDelegationReward = decimal.Zero
+	)
+
+	commissions, err := c.store.QueryBreathBlockRewardEvent(ctx, c.cfg.NodeRealOperator.Hex(), from.Unix(), to.Unix())
+	if err != nil {
+		log.Errorw("[cc] query commission error", "err", err)
+	}
+	for _, commission := range commissions {
+		commissionReward = commissionReward.Add(commission.Commission)
+	}
+	commissionReward = commissionReward.Div(decimal.NewFromInt(1e18))
+
+	selfDelegations, err := c.store.QueryDelegator(ctx, c.cfg.NodeRealOperator.Hex(),
+		c.cfg.NodeRealOperator.Hex(), from.Unix(), to.Unix())
+	if err != nil {
+		log.Errorw("[cc] query self delegation error", "err", err)
+	}
+	for _, selfDelegation := range selfDelegations {
+		selfDelegationReward = selfDelegationReward.Add(selfDelegation.Amount)
```

Comment:
Is `selfDelegation.Amount` the reward for each day? Or do we need to calculate the diff to get the reward for each day?

Reply:
fixed